### ISSUE TITLE
build instructions needed

### ISSUE THREAD

**Issue by SlrG** [veyon/veyon#674]
I have to apologize, but sadly I am unable to get the build process working myself.

- I installed docker and git, cloned the veyon source to get the .ci folder with the included dockerfiles.
- I changed into the linux.ubuntu.focal folder and used docker build, to create a development image which should have the needed dependencies installed.
- With docker run I started and changed into the image and cloned the veyon repo again, to have the source files and build scripts inside the image.
- Lastly I tried to build by running script.sh and while the process started (so everything should be setup correctly), it failed with errors.

What can I do, to get a working build? I need it to try and participate in the development by providing pull requests.
Shouldn't the dockerfile and script give a working build environment?
Maybe the steps to get it running could be described in the wiki, as I imagine other users might want a build setup, too?
Thank you very much for this great product. :+1:

**Comment by tobydox** [comment]
Depending on your system you don't even need Docker (which is mainly used for CI purposes only). Are you already working with Linux? If so, you can simply build Veyon as described in the README file, i.e. install all dependencies and open the project (CMakeLists.txt) in the Qt Creator IDE.

**Comment by SlrG** [comment]
Thank you for the quick answer. That got me a bit further I think. It sadly still fails to build however. Let me describe my process.

- download Focal Fossa https://releases.ubuntu.com/20.04/
- choose minimal installation
- choose update while installing
- sudo apt get install git
- clone veyon source
- install debian dependencies from README
- **Hint: need to use sudo!**
- **Problem: libqca2-dev not found!**
- install ubuntu focal dependencies used in the dockerfile
- **Problem: qt creator is not installed!**
- installed qt creator
- open CMakeLists.txt in qt creator
- try to build the project

It fails with the following CMakeError.log:
https://pastebin.com/Tk8VkqdN

I tried the same process on Debian and get the same errors.

What am I still missing or doing wrong? I'm very sorry, that I'm not able to solve this problem myself and hope it doesn't take too much of your time to reply.

**Comment by tobydox** [comment]
Thank you for your feedback! Indeed `libqca2-dev` does not exist any longer and has been replaced by the (also mentioned) `libqca-qt5-2-dev`. Since you're missing pthread files, you'll have to install `libc6-dev` which normally is installed automatically when installing `g++`. I updated the README.md file accordingly.

**Comment by SlrG** [comment]
Both packages are (and were already) installed, yet the build still fails. Qt Creator reports:
![qtcreator_veyon01](https://user-images.githubusercontent.com/978967/97228966-f1f98080-17d7-11eb-91f3-489f640377c2.png),
![qtcreator_veyon02](https://user-images.githubusercontent.com/978967/97228967-f32aad80-17d7-11eb-9327-9a0a51e98f31.png)
and
https://pastebin.com/hJc3cGJb

**Comment by tobydox** [comment]
It looks like you did not checkout the submodules properly, i.e. did not run `git clone --recursive ...` as described in the README. Please clone again and make sure to checkout to the 4.5 branch in order to work on the latest stable branch.

**Comment by SlrG** [comment]
Oh... That was a stupid mistake. Thank you very much for helping me get it running! Building the master tree does work now.

If I do "git checkout 4.5" however and try to build, I get the error, that the source file veyon/3rdparty/kldap/src/ber.cpp is missing. Indeed it is in veyon/3rdparty/kldap/src/core/ber.cpp instead. Am I still doing something wrong?

**Comment by tobydox** [comment]
Probably CMake/Qt Creator cached something. Please remove all build directories (build-veyon... in the parent directory) and try again.

**Comment by SlrG** [comment]
Sadly this does not work.

I deleted all build folders and the veyon source folder. I opened qtcreator and removed all projects and cleared the recent projects list and deleted the session storage. Now opening qtcreator gives no trace of veyon anymore.

Now I rebooted the vm to be sure and cloned the veyon repo with the command from the README again, which puts me inside the veyon folder where I do "git checkout 4.5". Afterwards I open qtcreator and click on open project, navigate to the veyon folder and open the CMakeLists.txt. The configuring process starts and exits with the error I posted above. Here it is again, as a screenshot:
![veyon-build-qtcreator1](https://user-images.githubusercontent.com/978967/97479637-bcc56d80-1952-11eb-8010-2b00a8361b96.png)
![veyon-build-qtcreator2](https://user-images.githubusercontent.com/978967/97479647-bfc05e00-1952-11eb-8f34-78bed45d774c.png)

Doing the same without the "git checkout 4.5" works fine and the configuration and the build process work.
